File: ristorino-backend/src/main/java/ubp/das/ristorinobackend/dto/restaurant/RestauranteDetalleDTO.java
package ubp.das.ristorinobackend.dto.restaurant;

import lombok.Data;
import ubp.das.ristorinobackend.dto.promotion.PromocionDTO;

import java.util.List;

@Data
public class RestauranteDetalleDTO {

    private Long id;
    private String nombre;
    private String tipoCocina;
    private String descripcion;
    private String imagenUrl;

    private List<SucursalDetalleDTO> sucursales;
    private List<PromocionDTO> promocionesVigentes;
}

File: ristoriniAngular/src/app/models/restaurante-detalle.dto.ts
import { PromocionDTO } from './promocion.dto';
import { SucursalDetalleDTO } from './sucursal-detalle.dto';

export interface RestauranteDetalleDTO {
  id: number;
  nombre: string;
  tipoCocina: string;
  descripcion: string;
  imagenUrl: string | null;
  sucursales: SucursalDetalleDTO[];
  promocionesVigentes: PromocionDTO[];
}